### PR TITLE
fix mergeObjects order

### PR DESCRIPTION
**Pull request by thefalked**
### Description

When using Arrays on a extended variant it was returning `[object object]`.

By changing the order of the function `mergeObjects` to first check if is an `array` it work as expected (joining the arrays).

---

### What is the purpose of this pull request?

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/jrgarciadev/tailwind-variants/blob/main/CONTRIBUTING.md).
- [x] Follow the [Style Guide](https://github.com/jrgarciadev/tailwind-variants/blob/main/CONTRIBUTING.md#style-guide).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).

### CODE REVIEW

File: src/__tests__/tv.test.ts [hunk 1]
```diff
@@ -2320,7 +2362,7 @@ describe("Tailwind Variants (TV) - Extends", () => {
         {
           isBig: true,
           color: "red",
-          class: "bg-red-500",
+          class: ["bg-red-500"],
```
Comment (mskelton) [inline src/__tests__/tv.test.ts:2365]: Why the change to the existing test? In general we should avoid changing existing tests.

File: src/__tests__/tv.test.ts [hunk 2]
```diff
@@ -2093,6 +2093,48 @@ describe("Tailwind Variants (TV) - Extends", () => {
...
+    const p = tv({
+      base: "text-base text-green-500",
+      variants: {
+        isBig: {
+          true: "text-5xl",
+          false: "text-2xl",
+        },
+        color: {
+          red: ["text-red-500 bg-red-100", "tracking-normal"],
+          blue: "text-blue-500",
```
Comment (mskelton) [inline src/__tests__/tv.test.ts:2106]: I think this test should not be just this case, but rather we should test all the combinations, e.g.

- extend: array, other: array
- extend: array, other: string
- extend: string, other: array
- extend: string, other: string

We can do this in the same test just using different variant values.

File: src/__tests__/tv.test.ts (hunk 1 above)
Comment (thefalked) [reply src/__tests__/tv.test.ts:2365]: I was just checking if the `compoundVariants` can also be used with array, but i can create new tests for this as well.

Comment (mskelton) [reply src/__tests__/tv.test.ts:2365]: This could be included in the one new test you added.

**Review by mskelton — APPROVED** [review]
